clients/merge-go-ethereum/geth.sh

```diff
@@ -81,7 +81,7 @@ if [ "$HIVE_NODETYPE" == "archive" ]; then
     FLAGS="$FLAGS --syncmode full --gcmode archive"
 fi
 if [ "$HIVE_NODETYPE" == "full" ]; then
-    FLAGS="$FLAGS --syncmode snap"
+    FLAGS="$FLAGS --syncmode full"
```

Comment:
This is probably wrong to do, but is there a recommended way for client to full sync but not be in archive mode?

Reply:
Why do you need to force full sync here?

Reply:
@fjl when I start a client in fast sync and begin propagating blocks to it, it discards them as it considers itself "syncing". I'm not sure how to resolve this, other than by forcing the node to be in full sync?

Reply:
I see. We don't have this problem in other tests, and I think it's because we initialize the nodes with at least one block. So what I can recommend is, just create a chain of one block using `cmd/hivechain` and initialize the eth1 node with it.

Reply:
Ah, okay. I will give that a try. Thanks.

Reply:
Hmm, unfortunately I realized that because the genesis is dynamic (`timestamp` need to be close to current time to be interoperable with the beacon chain). I suppose this could be worked around, but it seems like it may open up many edge cases if the network is on a different time than actual.

Reply:
What we could also do is, just launch another client to produce blocks, and have it propagate blocks to the merge client that is being tested. In geth, once a block has been imported via p2p, snap sync is considered done. Setting up an actual client to produce blocks might also remove the need to inject blocks from within the test code.

This reminds me, when I tried the eth2 simulator last time, it did PoW mining which made everything extremely slow. It would be so much better to use clique, and we do that in all other tests. Not sure if it is possible here because of TTD.

Reply:
Oh, very cool, I see you already kind of implemented my suggestion in https://github.com/ethereum/hive/pull/495/commits/661822d706f4386bf8464ac7bf15740c51503105.

Reply:
Yes, decided that the mock functionality isn't really being utilized yet so it makes more sense to simplify the PR. Clique would definitely be simpler - the main reason I've opted to do PoW was to as closely mimic the real transition as possible. Now that I'm not using the mock module, it seems the eth1 client just mines on 1 thread so maybe this light enough to be more acceptable? I also adjusted the parameters accordingly. If you still think Clique is preferable, I can update to using that instead.

I'll need to test if the snap sync is working now, but I'm not sure why it would've been different with the mock module because it also announced the block over devp2p. Maybe because the geth client had outstanding snap requests to the mock client, which didn't know how to respond to them?

Reply:
Okay switching to snap worked. Going to mark this as resolved - please reopen if you'd like me to using Clique instead of PoW.